src/Libraries/PythonNodeModelsWpf/PythonStringNode.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Dynamo.Controls;
+using Dynamo.Graph.Workspaces;
+using Dynamo.ViewModels;
+using Dynamo.Wpf;
+using PythonNodeModels;
+using PythonNodeModelsWpf.Controls;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+
+namespace PythonNodeModelsWpf
+{
+    class PythonStringNodeViewCustomization : VariableInputNodeViewCustomization, INodeViewCustomization<PythonStringNode>
+    {
+        private PythonStringNode pythonStringNodeModel;
+        private NodeView pythonStringNodeView;
+        private MenuItem pythonEngine2Item = new MenuItem { Header = PythonNodeModels.Properties.Resources.PythonNodeContextMenuEngineVersionTwo, IsCheckable = true };
+        private MenuItem pythonEngine3Item = new MenuItem { Header = PythonNodeModels.Properties.Resources.PythonNodeContextMenuEngineVersionThree, IsCheckable = true };
+
+        public void CustomizeView(PythonStringNode nodeModel, NodeView nodeView)
+        {
+            base.CustomizeView(nodeModel, nodeView);
+
+            pythonStringNodeModel = nodeModel;
+            pythonStringNodeView = nodeView;
+
+            // If it is a Debug build, display a python engine switcher
+            if (Dynamo.Configuration.DebugModes.IsEnabled("Python3DebugMode"))
+            {
+                var pythonEngineVersionMenu = new MenuItem { Header = PythonNodeModels.Properties.Resources.PythonNodeContextMenuEngineSwitcher, IsCheckable = false };
+                nodeView.MainContextMenu.Items.Add(pythonEngineVersionMenu);
+                pythonEngine2Item.Click += delegate { UpdateToPython2Engine(); };
+                pythonEngine3Item.Click += delegate { UpdateToPython3Engine(); };
```

Comment:
@SHKnudsen You may want to implement something similar to https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/PythonNodeModelsWpf/PythonNode.cs#L64 to unsubscribe these